### PR TITLE
Cart page refactor

### PR DESCRIPTION
**Pull request by dinohamzic**
## Description

This is a refactor and restyle of the `/cart` page and all its components.

## Motivation and Context

This is part of the ongoing refactor effort of the starter Solidus store.

## How Has This Been Tested?

We kept the old Solidus starter store tests and based the refactor on them.

## Screenshots (if appropriate):

Screen Shot 2020-03-20 at 09 53 34

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: app/views/spree/components/cart/_cart_items.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    collection: order.line_items,
+    as: :line_item,
+    locals: { order_form: order_form }
+  ) %>
+
+  <% if (
+    order.adjustments.nonzero.exists? ||
+    order.line_item_adjustments.nonzero.exists? ||
+    order.shipment_adjustments.nonzero.exists? ||
+    order.shipments.any?
```
Comment (kennyadsl) [inline app/views/spree/components/cart/_cart_items.html.erb:15]: I see this logic eligible for a partial's variable, like `is_order_empty`, WDYT?

File: app/views/spree/components/forms/inputs/_text.html.erb [hunk 2]
```diff
@@ -7,11 +7,13 @@
   label = false if local_assigns[:label].nil?
   min = false if local_assigns[:min].nil?
   placeholder = "" if local_assigns[:placeholder].nil?
+  small = false if local_assigns[:small].nil?
```
Comment (kennyadsl) [inline app/views/spree/components/forms/inputs/_text.html.erb:10]: I'm not really onboard with this: being small is something that works for our design but someone else could want to have that thing (also applies to button component) bigger. What about moving these rules to CSS only?

File: app/views/spree/components/cart/_cart_items.html.erb (hunk 1 above)
Comment (dinohamzic) [reply app/views/spree/components/cart/_cart_items.html.erb:15]: Great idea, much cleaner 👍

File: app/views/spree/components/forms/inputs/_text.html.erb (hunk 2 above)
Comment (dinohamzic) [reply app/views/spree/components/forms/inputs/_text.html.erb:10]: I'm not sure I'm following. `small` is toggle-able. If the user wants the button to be bigger (normal/default size), just don't pass anything when rendering the button component. 

Also, how would you go about moving the rule to CSS only?

Comment (kennyadsl) [reply app/views/spree/components/forms/inputs/_text.html.erb:10]: I mean that we can replace the `-small` definition from the HTML to the CSS. Something like (this is pseudo-code):

```
.coupon-code {
  .text-input {
    // code that make the button small
  }
}
```

In my perspective, this way, making the button smaller is a responsibility of the CSS only. If someone wants to make it bigger, it's a style/look&feel concern, they should not touch the HTML.

Comment (dinohamzic) [reply app/views/spree/components/forms/inputs/_text.html.erb:10]: The issue here is that if we need the button smaller in another place (we will be needing it in the checkout process), we would need to duplicate the css is several places.

Passing props that control the style (size, color, visibility etc.) is a common pattern in React for example. It avoids a lot of code duplication.

**Pull request merged**